release-23.1: sql: Exempt SET CLUSTER SETTING from throttling

spilchen opened the pull request:
Backport 1/1 commits from #131303.

/cc @cockroachdb/release

---

This change prevents the SET CLUSTER SETTING ... command from being throttled. Previously, it was subject to throttling, but since this command is used to install a new license and disable throttling, we want to ensure that the action itself is not throttled. This adds an exception to allow license updates.

This will be backported to versions 24.2, 24.1, 23.2, and 23.1.

Epic: CRDB-39988
Closes CRDB-42308
Release note: None
Release justification: This work is part of the CockroachDB core deprecation.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131742)
<!-- Reviewable:end -->

spilchen committed:
sql: Exempt SET CLUSTER SETTING from throttling

This change prevents the SET CLUSTER SETTING ... command from being
throttled. Previously, it was subject to throttling, but since this
command is used to install a new license and disable throttling, we want
to ensure that the action itself is not throttled. This adds an
exception to allow license updates.

This will be backported to versions 24.2, 24.1, 23.2, and 23.1.

Epic: CRDB-39988
Closes CRDB-42308
Release note: None
